library/psa_crypto.c

```diff
@@ -2365,19 +2365,27 @@ psa_status_t psa_mac_sign_finish( psa_mac_operation_t *operation,
      * unachievable MAC. */
     *mac_length = mac_size;
 
-    if( operation->id == 0 )
-        return( PSA_ERROR_BAD_STATE );
+    if( operation->id == 0 ) {
```

Comment:
Comment not for this code but for psa_mac/cipher_setup(). I think I'd prefer to do the same changes in those functions as well. That way if a user calls twice _setup() then the operation object is not valid anymore and not usable. Then in the negative tests when _setup() is called on an already active operation object, check that after the call it is inactive.